packages/run-protocol/src/vaultFactory/params.js

```diff
@@ -18,63 +18,107 @@ export const LIQUIDATION_MARGIN_KEY = 'LiquidationMargin';
 export const INTEREST_RATE_KEY = 'InterestRate';
 export const LOAN_FEE_KEY = 'LoanFee';
 
-/** @type {MakeElectorateParams} */
+/**
+ * @param {Amount} electorateInvitationAmount
+ */
 const makeElectorateParams = electorateInvitationAmount => {
   return harden({
     [CONTRACT_ELECTORATE]: makeGovernedInvitation(electorateInvitationAmount),
   });
 };
 
-/** @type {MakeLoanParams} */
-const makeLoanParams = (loanParams, rates) => {
+/**
+ * @param {LoanTiming} loanTiming
+ * @param {Rates} rates
+ */
+const makeLoanParams = (loanTiming, rates) => {
   return harden({
-    [CHARGING_PERIOD_KEY]: makeGovernedNat(loanParams.chargingPeriod),
-    [RECORDING_PERIOD_KEY]: makeGovernedNat(loanParams.recordingPeriod),
+    [CHARGING_PERIOD_KEY]: makeGovernedNat(loanTiming.chargingPeriod),
+    [RECORDING_PERIOD_KEY]: makeGovernedNat(loanTiming.recordingPeriod),
     [INITIAL_MARGIN_KEY]: makeGovernedRatio(rates.initialMargin),
     [LIQUIDATION_MARGIN_KEY]: makeGovernedRatio(rates.liquidationMargin),
     [INTEREST_RATE_KEY]: makeGovernedRatio(rates.interestRate),
     [LOAN_FEE_KEY]: makeGovernedRatio(rates.loanFee),
   });
 };
 
-/** @type {MakeVaultParamManager} */
-const makeVaultParamManager = (loanParams, rates) => {
-  // @ts-ignore It's a VaultParamManager
+/**
+ * @param {LoanTiming} initialValues
+ * @returns {ParamManagerFull & {
+ *   updateChargingPeriod: (period: bigint) => void,
+ *   updateRecordingPeriod: (period: bigint) => void,
+ * }}
+ */
+const makeLoanTimingManager = initialValues => {
+  // @ts-expect-error until makeParamManagerBuilder can be generic */
+  return makeParamManagerBuilder()
+    .addNat(CHARGING_PERIOD_KEY, initialValues.chargingPeriod)
+    .addNat(RECORDING_PERIOD_KEY, initialValues.recordingPeriod)
+    .build();
+};
+
+/**
+ * XXX maybe call this "loan terms"?
```

Comment:
to whom is this `XXX` comment directed? Why call it "loan terms"? Is there something wrong with the current name? if so, what?

Reply:
The XXX is directed at whoever is reading this code, but I think this one can be resolved before merging.

I thought "terms" instead of "rates" because while it includes `interestRate` it also has `initialMargin`, `liquidationMargin` and `loanFee`. The latter don't seem like rates to me but they are part of the terms of the loan.

I didn't make the change because I thought maybe the charging and recording period could be considered terms of the loan… but since those are parameters of the whole vault factory they aren't part of any particular loan.

So I think Terms is better than Rates. But I'm not as sure about the "Loan" term (pun intended) because vaults, while they share some mechanics with loans, are explicitly not loans. Maybe ChargingTerms?

Reply:
I have some hesitation about changing this name, as it's not required to address #4185 . It's a bit of a "drive-by." Plus, I feel a little like when my spouse asks me whether I like the new paint color; I just don't have an opinion.

OTOH, @dtribble did ask you to look into naming issues such as this. See if you can grab him briefly?

For somewhat arbitrary issues like this, I'm kinda inclined to fall back on an executive hierarchy (#3382); that is: talk about it with @Chris-Hibbert .

I guess if it's between you and me: I'll agree to a change if you think it's sufficiently worthwhile to add a test :)
Is it part of the package's external API? If so, that makes it a breaking change, yes? So it would need its own conventional commit to get it in the CHANGELOG. (or do we use NEWS.md in this package?)

Reply:
Good call that it's out of scope. I've cut it and will look into it separately.

I don't think it wouldn't change the external API because Agoric controls the governance of the loan time periods.